hadoop-hdds/common/src/main/java/org/apache/hadoop/hdds/ratis/RatisUpgradeUtils.java

```diff
@@ -78,14 +82,26 @@ public static void waitForAllTxnsApplied(
       throw new IOException("Index from Snapshot does not match last applied " +
           "Index");
     }
+
+    if (purgeLogsAfter) {
+      CompletableFuture<Long> purgeFuture =
+          impl.getState().getLog().purge(snapshotIndex);
+      try {
+        Long purgeIndex = purgeFuture.get();
```

Comment:
I prefer to have a timeout limitation here, it will wait forever if it hasn't finished the purge behavior.
Can we reuse maxTimeToWaitSeconds as the purge timeout and catch timeout exception?
And then add a corresponding test case for this corner case.